packages/core/src/test/amazonqFeatureDev/controllers/chat/controller.test.ts

```diff
@@ -385,22 +398,113 @@ describe('Controller', () => {
+                            sendAnswerSpy.calledWith({
+                                type: 'answer',
+                                tabID,
+                                message: error.message + messageWithConversationId(session?.conversationIdUnsafe),
+                                canBeVoted: true,
+                            })
+                        )
+                        break
+                    case MonthlyConversationLimitError.name:
+                        assert.ok(sendMonthlyLimitErrorSpy.calledWith(tabID))
+                        break
+                    case FeatureDevServiceError.name:
+                    case UploadCodeError.name:
+                    case UserMessageNotFoundError.name:
+                    case TabIdNotFoundError.name:
+                    case PrepareRepoFailedError.name:
+                        assert.ok(
+                            sendErrorMessageSpy.calledWith(
+                                createTestErrorMessage(error.message),
+                                tabID,
+                                session?.retries,
+                                session?.conversationIdUnsafe
+                            )
+                        )
+                        break
+                    case PromptRefusalException.name:
+                    case ZipFileError.name:
+                        assert.ok(
+                            sendErrorMessageSpy.calledWith(
+                                createTestErrorMessage(error.message),
+                                tabID,
+                                0,
+                                session?.conversationIdUnsafe,
+                                true
+                            )
+                        )
+                        break
+                    case NoChangeRequiredException.name:
+                    case CodeIterationLimitError.name:
+                    case UploadURLExpired.name:
+                        assert.ok(
+                            sendAnswerSpy.calledWith({
+                                type: 'answer',
+                                tabID,
+                                message: error.message,
+                                canBeVoted: true,
+                            })
+                        )
+                        break
+                    default:
+                        assert.ok(
+                            sendErrorMessageSpy.calledWith(
+                                i18n('AWS.amazonq.featureDev.error.codeGen.default'),
+                                tabID,
+                                session?.retries,
+                                session?.conversationIdUnsafe,
+                                true
+                            )
+                        )
+                        break
+                }
+            }
+
+            runs.forEach((run) => {
+                it(`should handle ${run.name}`, async function () {
```

Comment:
For reference this is one way of de-duplicating tests, but it would also be fine to have a bunch of manally-declared it() blocks which call a common function.